### PR TITLE
fix: typescript support

### PR DESCRIPTION
**Pull request by vkuprin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,11 +20,11 @@
   "exports": {
     "./react": {
       "require": "./dist/cjs/react-vt.js",
-      "import": "./src/react-vt.js"
+      "import": "./src/react-vt.ts"
```
Comment (FilipChalupa) [inline package.json]: This will break projects written in JavaScript without TypeScript, right?

Comment (vkuprin) [reply package.json]: Yes, I'm suggesting that we follow semantic versioning. Anyone who wants to upgrade will see that the major version in package.json of use-view-transition has changed

File: src/next-vt.tsx [hunk 2]
```diff
@@ -1,56 +1,64 @@
...
         };
     }, []);
 }
 
-function RouterEventsNotifier() {
-  usePathname();
-  useSearchParams();
-  return null;
+function RouterEventsNotifier(): null {
+    // usePathname();
```
Comment (noamr) [inline src/next-vt.tsx]: If you want to erase it erase rather than comment out.

File: src/react-vt.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            return;
+
+        function captureClick(event: MouseEvent) {
+            const target = event.target as Element;
+            if (!target.matches(match) || !event.isTrusted)
+                return;
+
+            event.preventDefault();
+            event.stopPropagation();
+            // @ts-ignore
```
Comment (noamr) [inline src/react-vt.ts]: ???

File: src/react-vt.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+interface ViewTransitionController {
+    transitionState: TransitionState;
+    startViewTransition: (domUpdateCallback?: () => PromiseLike<void>) => PromiseLike<void> | void;
+    suspendViewTransitionCapture: typeof suspendViewTransitionCapture;
+    resumeViewTransitionCapture: typeof resumeViewTransitionCapture;
+}
+
+export function useViewTransition(): ViewTransitionController {
+    const [transitionState, setTransitionState] = useState<TransitionState>("idle");
+    // @ts-ignore
```
Comment (noamr) [inline src/react-vt.ts]: ???

File: src/react-vt.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            setTransitionState("animating");
+        }).catch(e => {
+            console.error(e)
+            setTransitionState("skipped");
+        });
+    }
+
+    return {
+        transitionState,
+        // @ts-ignore
```
Comment (noamr) [inline src/react-vt.ts]: ???

File: src/next-vt.tsx (hunk 2 above)
Comment (vkuprin) [reply src/next-vt.tsx]: Okey, please check last patch

File: src/react-vt.ts (hunk 3 above)
Comment (vkuprin) [reply src/react-vt.ts]: Okey, please check last patch

File: package.json (hunk 1 above)
Comment (FilipChalupa) [reply package.json]: I mean it makes it impossible to make new projects without TypeScript.

**Review by FilipChalupa — APPROVED** [review]

**Review by noamr — APPROVED** [review]

**Pull request merged**